### PR TITLE
[#12048] Add Section and Team entity

### PR DESCRIPTION
**Pull request by samuelfangjw**
Part of #12048

### CODE REVIEW

File: src/main/java/teammates/storage/sqlentity/Section.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    private List<Team> teams;
+
+    @CreationTimestamp
+    @Column(nullable = false, updatable = false)
+    private Instant createdAt;
+
+    @UpdateTimestamp
+    @Column(nullable = false)
+    private Instant updatedAt;
+
```
Comment (daongochieu2810) [inline src/main/java/teammates/storage/sqlentity/Section.java:51]: is it possible to move these to `BaseEntity`?

File: src/main/java/teammates/storage/sqlentity/BaseEntity.java [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
 import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Column;
 import jakarta.persistence.Converter;
+import jakarta.persistence.MappedSuperclass;
 
 /**
  * Base class for all entities.
  */
+@MappedSuperclass
 public abstract class BaseEntity {
```
[the remark this answers is not in the document]
Comment (samuelfangjw) [reply src/main/java/teammates/storage/sqlentity/BaseEntity.java:20]: no, id field is not the same across entities

**Review by daongochieu2810 — APPROVED** [review]

**Review by domlimm — APPROVED** [review]
LGTM!

Just curious, should we have the toString method in these entities?

**Comment by samuelfangjw** [comment]
> LGTM!
> 
> Just curious, should we have the toString method in these entities?

good catch, should add!

**Pull request merged**